x-pack/plugins/maps/public/connected_components/mb_map/tile_status_tracker.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Map as MapboxMap, MapDataEvent } from 'mapbox-gl';
+import _ from 'lodash';
+import { ILayer } from '../../classes/layers/layer';
+import { SPATIAL_FILTERS_LAYER_ID } from '../../../common/constants';
+
+interface MbTile {
+  // references internal object from mapbox
+  aborted?: true;
+}
+
+interface Tile {
+  mbKey: string;
+  mbSourceId: string;
+  mbTile: MbTile;
+}
+
+export class TileStatusTracker {
+  private _tileCache: Tile[] | null;
```

Comment:
Why can _tileCache be null? How about just allowing Tile[] since its set to an empty array in the constructor. This would avoid logic for checking for null every time _tileCache is used.

Reply:
so we can remove the reference in the destroy-handler and make available for GC. I can change this to clear the array instead to keep typing cleaner.